moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -216,8 +217,35 @@ const NewProjectMap = ({
 
     if (!layerName || !getClickEditableLayerNames().includes(layerName)) return;
 
-    const clickedFeatureId = getFeatureId(e.features[0], layerName);
-    const selectedFeature = getGeoJSON(e);
+    let selectedFeature = getGeoJSON(e);
+    const selectedFeatureId = getFeatureId(selectedFeature, layerName);
+
+    if (
+      selectedFeature.geometry.type === "LineString" ||
+      selectedFeature.geometry.type === "MultiLineString"
+    ) {
+      // Query features in the current viewport to identify any that have the same
+      // ID has the selected feature. This is an artifact of the tiling process.
+      // Split features must be re-combined before saving to state (and DB).
+      // We are making the assumption that any split features would be visible in the
+      // map viewport when the feature is clicked. This may not be a safe assumption?
+      // The alternative is to supply our own search geometry bbox
+      const renderedFeatures = mapRef.current.queryRenderedFeatures();
+
+      // It's possible to pass a filter function to queryRenderedFeatures. We
+      // could theoretically filter for features that match the ID of interest, but
+      // I could not get the expression to work. I doubt the performance gain would
+      // be significant vs our own filter call.
+      const splitFeatures = renderedFeatures.filter(
+        feature => getFeatureId(feature, layerName) === selectedFeatureId
+      );
```

Comment:
@johnclary Ah, this is a gem! 💯 💎 It didn't occur to me to search the rest of the feature like this. The bbox would help performance for sure.